### PR TITLE
api: Use 'note' constructs for API documentation.

### PR DESCRIPTION
**Pull request by aniemetz**
This uses `@note` for notes in the C++ API documentation, `.. note::`
for Python, and `@apiNote` for Java.

### CODE REVIEW

File: src/api/cpp/cvc5.h [hunk 1]
```diff
@@ -987,9 +991,10 @@ class CVC5_EXPORT Op
 
   /**
    * The internal node wrapped by this operator.
-   * Note: This is a shared_ptr rather than a unique_ptr to avoid overhead due
-   *       to memory allocation (cvc5::Node is already ref counted, so this
-   *       could be a unique_ptr instead).
+   *
+   * @note This is a ``std::shared_ptr`` rather than a ``std::unique_ptr`` to
+   *       avoid overhead due to memory allocation (``cvc5::Node`` is already
+   *       ref counted, so this could be a ``std::unique_ptr`` instead).
```
Comment (nafur) [inline src/api/cpp/cvc5.h:997]: ... so it should be changed to a unique_ptr?

Comment (aniemetz) [reply src/api/cpp/cvc5.h:997]: No, because overhead due to memory allocation, as the note says.

File: src/api/python/cvc5.pxi [hunk 2]
```diff
@@ -3060,19 +3062,23 @@ cdef class Term:
...
+        .. code::
+
+            (union
+                (singleton c1) ... (union (singleton c_{n-1}) (singleton c_n))))
+
+        where ``c1 ... cn`` are values ordered by id such that
+        ``c1 > ... > cn`` (see also :cpp:func:`cvc5::api::Term::operator>()`).
+
+        .. note::
+            A universe set term `(kind SET_UNIVERSE)` is not considered to be
```
Comment (nafur) [inline src/api/python/cvc5.pxi:3077]:
```suggestion
            A universe set term ``(kind SET_UNIVERSE)`` is not considered to be
```

**Review by nafur — APPROVED** [review]
LGTM. I've not checked every single place in the generated docs, though.

**Comment by aniemetz** [comment]
> LGTM. I've not checked every single place in the generated docs, though.

I have and unless I missed anything, it should be generated correctly.

**Pull request merged**